src/utils/daos_hdlr.c

```diff
@@ -539,6 +543,16 @@ file_chmod(struct cmd_args_s *ap, struct file_dfs *file_dfs, const char *path,
 		}
 	} else if (file_dfs->type == DAOS) {
 		rc = dfs_sys_chmod(file_dfs->dfs_sys, path, mode);
+		if (rc == ENOTSUP && !ignore_unsup) {
```

Comment:
Im nor sure i quite understand why a retry is needed here.
so the user has not requested ignore_unsup, but you still want to ignore_unsup?

Reply:
Yeah, the idea is...

with ignore_unsup:
- unset the bits before the call to chmod
- there will be no error, we won't log a warning, and we exit 0

without ignore_unsup:
- the call to chmod fails
- unset the bits and try again
- print a warning and exit non-zero

Some history of the discussion in the bug: https://daosio.atlassian.net/browse/DAOS-10942

Reply:
Looks like with and without `--ignore-unsupported`, the unsupported mode bits are removed. By default, `num_chmod_enotsup` is reported at the end. But with `--ignore-unsupported` it's silent.

Maybe to avoid a retry we could do something like this upfront?
```
if (!ignore_unsup && mode & (S_ISVTX | S_ISGID | S_ISUID))
    (*num_chmod_enotsup)++;
mode &= ~(S_ISVTX | S_ISGID | S_ISUID);
```

Reply:
That seems reasonable. The only downside I can see is that maybe we eventually support these bits and it's not an error anymore. In that case we'd probably want to update this code anyway though.

Reply:
Yeah, I think we'd want to update this anyway if we support in the future, and if at all I expect it would be far off

Reply:
if you are going to update this please also merge latest master to your branch as it is lagging behind a bit now.

Reply:
Shoot, my attempt to rebase brought in a bunch of random commits. Sorry, I'm not used to github yet.

Reply:
The flow is usually
```
git checkout my_branch
git fetch
git merge origin/master
```

Reply:
Yeah, I'm talking to Mike and Jeff and they are saying that github doesn't handle rebases well. Okay let me see if I can fix it. If not, I'll create a new PR.

Reply:
Did you use `rebase` or `merge`? `rebase` is trickier because it re-writes history (i.e. commits end up with different hashes). It's better to use `merge` after people have started reviewing so the history of commits is preserved.
Another common thing is if you do `git merge master`, your local `master` might not be up-to-date with the remote `master`. So you either have to do this
```
git checkout master
git fetch
git pull
git checkout my_branch
git merge master
```
or (simpler I think)
```
git checkout my_branch
git fetch
git merge origin/master
```

Reply:
I did a rebase the first time. I think it should be okay now.